### PR TITLE
Ignore -fPIC flag when not building relocatable code

### PR DESCRIPTION
**Pull request by sbc100**
This issue was supposed to have been fixed in lld but it seems that
issues remain.  See https://reviews.llvm.org/D65922.

Fixes: #9690, #9013

### CODE REVIEW

**Comment by sbc100** [comment]
PTAL

File: tests/test_core.py [hunk 1]
```diff
@@ -8359,6 +8359,11 @@ def test_undefined_main(self):
     '''
     self.build(src, self.get_dir(), 'test.c')
 
+  def test_fpic_static(self):
+    self.emcc_args.append('-fPIC')
+    self.emcc_args.remove('-Werror')
+    self.do_run_in_out_file_test('tests', 'core', 'test_hello_world')
```
[the remark this answers is not in the document]
Comment (sbc100) [reply tests/test_core.py]: This test fails withou the corresponding change to strip the `-fPIC`.   I just reconfirmed.   There might be multiple bugs but the emcc.py change makes this test start passing.

File: emcc.py [hunk 2]
```diff
@@ -1883,6 +1883,9 @@ def is_link_flag(flag):
         return any(flag.startswith(x) for x in ('-l', '-L', '-Wl,'))
 
       compile_args = [a for a in newargs if a and not is_link_flag(a)]
+      if '-fPIC' in compile_args and not shared.Settings.RELOCATABLE:
+        shared.warning('ignoring -fPIC flag when not building with SIDE_MODULE or MAIN_MODULE')
+        compile_args.remove('-fPIC')
```
[the remark this answers is not in the document]
Comment (sbc100) [reply emcc.py]: Yes and yes. 

For now, until we fix the underlying issue with static linking of PIC objects, then we can allow `-fPIC` to control this again.

File: tests/test_core.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (kripken) [reply tests/test_core.py]: Ah ok, sounds good on the test then.

File: emcc.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kripken) [reply emcc.py]: Sgtm, but (separately from this PR if you want) please update the docs to mention that with the new backend, MAIN/SIDE_MODULES must have their source files built with `MAIN/SIDE_MODULE`. E.g. here in the list of big backend differences, https://emscripten.org/docs/compiling/WebAssembly.html#backends

**Comment by kripken** [comment]
Please add this to the docs (that with the new backend, one must add flags a compile time too). And also this is a breaking change for some users (as `-fPIC` used to work at compil;e time, and not wont, for SIDE/MAIN_MODULE users), please mention that in the changelog.

**Pull request merged**